protocols/relay/src/behaviour.rs

```diff
@@ -171,14 +171,14 @@ pub enum Event {
     /// Accepting an inbound reservation request failed.
     ReservationReqAcceptFailed {
         src_peer_id: PeerId,
-        error: inbound_hop::UpgradeError,
+        error: inbound_hop::Error,
     },
```

Comment:
@mxinden Should we include the `ConnectionId` here so that users _could_ close the connection if they wanted to?

Reply:
:+1: sounds good. For the sake of consistency, how about including it in every event variant?

Reply:
Is it worth the complexity though? The more I am working through the codebase and thinking about the error cases, the less I am convinced that we should emit errors for something like that at all.

In this particular case, all of these errors are essentially protocol violations by the remote peer. I wouldn't know what code to write in response to such an error. If all streams fail, the keep alive algorithm will close the connection anyway.

The only case I can think of that maybe justifies keep these events are the metrics. But the value of those is also questionable. Why do I need _metrics_ on how many "broken" requests I am getting? This is what logs are for, no? A node operator can (should?) create alerts on certain logs. These errors are certainly unexpected so perhaps we should just log them as `warn!` and be done with it?

Reply:
I opted for deprecating them for now and open an issue where users can weigh in if they need the events for something else. If nobody speaks up until the next breaking release, I'd suggest to move forward with removing them.

See https://github.com/libp2p/rust-libp2p/issues/4757.

Reply:
> Why do I need _metrics_ on how many "broken" requests I am getting? This is what logs are for, no? A node operator can (should?) create alerts on certain logs. These errors are certainly unexpected so perhaps we should just log them as `warn!` and be done with it?

I would expect users to set up alerting via metrics, not logs.

I do believe metrics on errors are important. E.g. when error rates suddenly spike, that is worth alerting on.

Agreed that exposing the errors just for `libp2p-metrics` is unfortunate. Long term I would like to integrate `prometheus-client` into `libp2p-*` directly.

Reply:
> > Why do I need _metrics_ on how many "broken" requests I am getting? This is what logs are for, no? A node operator can (should?) create alerts on certain logs. These errors are certainly unexpected so perhaps we should just log them as `warn!` and be done with it?
> 
> I would expect users to set up alerting via metrics, not logs.
> 
> I do believe metrics on errors are important. E.g. when error rates suddenly spike, that is worth alerting on.

If we'd follow that logic, we shouldn't be logging any errors or warnings, right?

I don't think that solution scales. I am no SRE but it sounds painful to configure alerts based on many different metrics when the alternative is as easy as: "Alert me if there is a sudden increase of `warn` logs"?

As I've argued in the issue, I think metric based alerts are super powerful if you are tracking those metrics anyway. But just creating them so you can set an alert when they show up seems like a lot of unnecessary effort, both for us as maintainers and for users.

Can we merge it with the warnings or are you fundamentally opposed to it?